compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/optimize-ssr.expect.md

```diff
@@ -0,0 +1,32 @@
+
+## Input
+
+```javascript
+// @enableOptimizeForSSR
+function Component() {
+  const [state, setState] = useState(0);
+  const ref = useRef(null);
+  const onChange = e => {
+    setState(e.target.value);
+  };
+  useEffect(() => {
+    log(ref.current.value);
```

Comment:
This is where the compiler machinery helps: the ref is referenced here, but DCE runs in a fixpoint. By pruning the effect call that makes its callback dead code, which then removes this reference to the ref. Then the OptimizeForSSR pass has already pruned the ref prop, so we're able to remove the useRef.